app/controllers/likes_controller.rb

```diff
@@ -2,6 +2,12 @@ class LikesController < ApplicationController
   def create
     @post = Post.find(params[:post_id])
     @like = @post.likes.create(author: current_user, post: @post)
-    redirect_to user_post_path(User.find(params[:user_id]), @post)
+    respond_to do |format|
+      if @post
+        format.html do
+          redirect_to user_post_path(User.find(params[:user_id]), @post), notice: 'You have liked the post!'
+        end
+      end
```

Comment:
- [ ] Let's also add a flash message for a not successful path. Kindly add a flash message for when the like is not successfully created.